File: TreasurersApp/ClientApp/src/app/models/ContributionTypeActionResult.ts
import { ContributionType } from './ContributionType';
import { ContributionTypeService } from '../maintenance/contribution-type-maintenance/contribution-type.service';

export class ContributionTypeActionResult {
  success = false;
  statusMessages: string[] = [];
  contributionType: ContributionType = null;

  constructor(private contributionTypeService: ContributionTypeService) {
    this.success = false;
    this.contributionType = this.contributionTypeService.newContributionType();
    this.statusMessages = [];
  }
}

File: TreasurersApp/Models/ContributionTypeActionResult.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TreasurersApp.Models
{
    public class ContributionTypeActionResult : WebResult<ContributionType>
    {
        public ContributionTypeActionResult(bool success, List<string> statusMessages, ContributionType contributionType)
            :base(success, statusMessages, contributionType)
        {
        }
    }
}
